irohad/ametsuchi/impl/redis_flat_block_query.cpp

```diff
@@ -111,5 +127,54 @@ namespace iroha {
           });
     }
 
+    rxcpp::observable<model::Transaction> RedisFlatBlockQuery::getTxByHash(
+        std::string hash) {
+      return rxcpp::observable<>::create<model::Transaction>(
+          [this, hash](auto subscriber) {
+            auto blockId = this->getBlockId(hash);
+            if (not blockId) {
+              subscriber.on_completed();
+              return;
+            }
+
+            block_store_.get(blockId.value()) | [](auto bytes) {
+              return model::converters::stringToJson(bytesToString(bytes));
+            } | [this](const auto &json) {
+              return serializer_.deserialize(json);
+            } | [&](const auto &block) {
+              for (auto &tx : block.transactions) {
```

Comment:
Use `boost::for_each` and `boost::adaptor::filtered` instead

Reply:
I do not think we should rewrite everything to boost now and I'm not sure the resulted code will be easier to read. Why do you think it should be rewritten?

Reply:
The general reason is no reasons to write for cycles with business logic inside. Better to use composable functions like for_each and filter. 
Another reason is using of methods clarify code, because handling moving into one independent function( or lambda).
And last, here code written in functional style, better to follow one paradigm if possible.